src/com/myorg/myui5lib/Example.ts

```diff
@@ -1,65 +1,64 @@
-import ExampleRenderer from "./ExampleRenderer";
-import { ExampleColor } from "./library";
-import type { MetadataOptions } from "sap/ui/core/Element";
-
-
-/**
- * Constructor for a new <code>com.myorg.myUI5Library.Example</code> control.
- *
- * Some class description goes here.
- * @extends Control
- *
- * @author OpenUI5 Team 
- * @version ${version}
- *
- * @constructor
- * @public
- * @name com.myorg.myUI5Library.Example
- */
-export default class Example extends Control {
-
-	// The following three lines were generated and should remain as-is to make TypeScript aware of the constructor signatures
-	constructor(id?: string | $ExampleSettings);
-	constructor(id?: string, settings?: $ExampleSettings);
-	constructor(id?: string, settings?: $ExampleSettings) { super(id, settings); }
-
-	static readonly metadata: MetadataOptions = {
-		library: "com.myorg.myUI5Library",
-		properties: {
-			/**
-			 * The text to display.
-			 */
-			text: {
-				type: "string",
-				group: "Data",
-				defaultValue: null
-			},
-			/**
-			 * The color to use (default to "Default" color).
-			 */
-			color: {
-				type: "com.myorg.myUI5Library.ExampleColor",
-				group: "Appearance",
-				defaultValue: ExampleColor.Default
-			}
-		},
-		events: {
-			/**
-			 * Event is fired when the user clicks the control.
-			 */
-			press: {}
-		}
-	};
-
-	static renderer = ExampleRenderer;
-
-	onclick = () => {
-		this.firePress();
-	}
-}
\ No newline at end of file
+/*!
+ * ${copyright}
+ */
+
+// Provides control com.myorg.myui5lib.Example.
+import Control from "sap/ui/core/Control";
+import ExampleRenderer from "./ExampleRenderer";
+import { ExampleColor } from "./library";
+import type { MetadataOptions } from "sap/ui/core/Element";
+
+
+/**
+ * Constructor for a new <code>com.myorg.myui5lib.Example</code> control.
+ *
+ * Some class description goes here.
+ * @extends Control
+ *
+ * @author OpenUI5 Team
+ * @version ${version}
+ *
+ * @constructor
+ * @public
+ * @name com.myorg.myui5lib.Example
+ */
+export default class Example extends Control {
+	// The following three lines were generated and should remain as-is to make TypeScript aware of the constructor signatures
+	constructor(id?: string | $ExampleSettings);
+	constructor(id?: string, settings?: $ExampleSettings);
+	constructor(id?: string, settings?: $ExampleSettings) { super(id, settings); }
+
+	static readonly metadata: MetadataOptions = {
+		library: "com.myorg.myui5lib",
+		properties: {
+			/**
+			 * The text to display.
+			 */
+			text: {
+				type: "string",
+				group: "Data",
+				defaultValue: null
+			},
+			/**
+			 * The color to use (default to "Default" color).
+			 */
+			color: {
+				type: "com.myorg.myui5lib.ExampleColor",
+				group: "Appearance",
+				defaultValue: ExampleColor.Default
+			}
+		},
+		events: {
+			/**
+			 * Event is fired when the user clicks the control.
+			 */
+			press: {}
+		}
+	};
+
+	static renderer: typeof ExampleRenderer = ExampleRenderer;
```

Comment:
Is the `typeof ExampleRenderer` only needed because of the export changes? And could they be undone? It's a bit unfortunate/strange that this has to be written.

Reply:
We can maybe check again, but I doubt that this works in all scenarios - at least I exactly tried this and there were issues. But if you have time to recheck, then we just need to adopt the places where we prevent the export default. Thanks...

Reply:
@akudev: One more hint which speaks for keeping it as is: `ts-typecheck` works fine! This is only relevant when the `ui5-tooling-transpile` creates the type defintions by converting the modules to *ambient modules*. So, maybe I can workaround this in the `ui5-tooling-transpile` but I would keep this for now and adopt it later once I found a solution!

Reply:
https://github.com/colinhacks/zod/issues/1040